File: 0206-reverse-linked-list/0206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head){
    struct ListNode * prev = NULL;
    struct ListNode * curNode = head;
    struct ListNode * next = NULL;
    while (curNode != NULL){
        next = curNode->next;
        curNode->next = prev;
        prev = curNode;
        curNode = next;
    }
    return prev;        
}   

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        node, prev = head, None
        
        while node:
            next, node.next = node.next, prev # node이동, node.next = prev로 연결
            prev, node = node, next #다음 step을 위해서 prev 이동, node 이동
            
        return prev